src/pretty/list.rs

```diff
@@ -0,0 +1,276 @@
+use pretty::BoxDoc;
+use typst_syntax::{ast::*, SyntaxKind, SyntaxNode};
+
+use crate::util::FoldStyle;
+
+use super::{comment::comment, util::is_comment_node, PrettyPrinter};
+
+pub enum CommentedItem<'a> {
+    Comment(BoxDoc<'a, ()>),
+    Commented(BoxDoc<'a, ()>, BoxDoc<'a, ()>),
+}
```

Comment:
Please consider use a more descriptive name for enum variants. You may also want to use enum structs to assign names for each field. This should be better than unnamed tuples.